dali/pipeline/operators/reader/loader/coco_loader.cc

```diff
@@ -301,12 +301,13 @@ void ParseAnnotationFilesHelper(std::vector<std::string> &annotations_filename,
     DALI_ENFORCE(f, "Could not open JSON annotations file");
     f.seekg(0, std::ios::end);
     size_t file_size = f.tellg();
-    std::unique_ptr<char, std::function<void(char*)>> buff(new char[file_size],
-                          [](char* data) {delete [] data;});
+    std::vector<char> buff;
```

Comment:
```suggestion
    std::vector<char> buff(file_size + 1);
```
and remove the next line

Reply:
I though that `resize` doesn't value initialization but it does in the same way as bugg(n). As json could have 0.5GB I want to avoid that. Reverting back to `unique_ptr`.